services/datamanager/builtin/capture_test.go

```diff
@@ -247,18 +249,26 @@ func TestSwitchResource(t *testing.T) {
 	})
 	test.That(t, err, test.ShouldBeNil)
 
+	initialData, err := getSensorData(captureDir)
+	test.That(t, err, test.ShouldBeNil)
+
 	passTimeCtx2, cancelPassTime2 := context.WithCancel(context.Background())
 	donePassingTime2 := passTime(passTimeCtx2, mockClock, captureInterval)
 
 	// Test that sensor data is captured from the new collector.
-	waitForCaptureFilesToExceedN(captureDir, len(getAllFileInfos(captureDir)))
+	waitForCaptureFilesToExceedNFiles(captureDir, len(getAllFileInfos(captureDir))+10)
```

Comment:
Why 10?

Reply:
I had this in just for testing and removed it in a later commit, you may have been looking at outdated code!

Reply:
I 100% was lol